src/hooks/usePollOffchainMessage.ts

```diff
@@ -0,0 +1,52 @@
+import { useEffect, useState } from 'react'
+import { safeMsgDispatch, SafeMsgEvent } from '@/services/safe-messages/safeMsgEvents'
+import { getSafeMessage, type SafeMessage } from '@safe-global/safe-gateway-typescript-sdk'
+import { Errors, logError } from '@/services/exceptions'
+
+const usePollOffchainMessage = (chainId: string, safeMessageHash: string, onClose: () => void, requestId?: string) => {
+  const [ongoingMessage, setOngoingMessage] = useState<SafeMessage>()
+  const [isPolling, setIsPolling] = useState(false)
+  const CONFIRMATIONS_POLLING_INTERVAL = 3000
+
+  useEffect(() => {
```

Comment:
We could use `useAsync` here and poll as we do for Safe info (with an interval counter).

Reply:
After our discussion: we need not poll here, we already have the messages being polled for. We can reference the store instead.

Reply:
I removed the single message polling